pkg/network/server.go

```diff
@@ -321,7 +330,10 @@ func (s *Server) Shutdown() {
 		s.notaryRequestPool.StopSubscriptions()
 	}
 	close(s.quit)
+	<-s.broadcastTxFin
+	<-s.runProtoFin
```

Comment:
OK, and the last comment: how about waiting for `s.runFin`? I'm sure this code was here several force-pushes ago.